src/components/LocationFetcher.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div></div>
+</template>
+
+<script>
+import { getLabwhereLocations } from '@/services/labwhere/client.js'
+import { getCoordinateForLabware } from '@/services/labwhere/helpers.js'
+
+export default {
+  name: 'LocationFetcher',
+  props: {
+    barcodes: {
+      type: Array,
+      required: true,
+    },
+  },
+  emits: ['locationData'],
+  watch: {
+    barcodes: {
+      immediate: true,
+      handler() {
+        this.fetchLocations()
+      },
+    },
+  },
+  methods: {
+    async fetchLocations() {
+      if (this.barcodes.length === 0) {
+        this.$emit('locationData', this.defaultLocations())
+        return
+      }
+
+      try {
```

Comment:
These methods look like they belong elsewhere maybe in the helper or labwhere client ...